src/lib/storage/table.cpp

```diff
@@ -17,53 +19,103 @@
+  auto segment = make_shared_by_data_type<BaseSegment, ValueSegment>(type);
+  _chunks.front()->add_segment(segment);
 }
 
 void Table::append(std::vector<AllTypeVariant> values) {
-  // Implementation goes here
-}
+  // Check if the latest Chunk has already reached maximum size and create a new one if necessary
+  if (this->is_new_chunk_needed()) {
+    _chunks.push_back(std::make_shared<Chunk>());
 
-uint16_t Table::column_count() const {
-  // Implementation goes here
-  return 0;
+    for (auto type_name : _column_types) {
+      auto segment = make_shared_by_data_type<BaseSegment, ValueSegment>(type_name);
+      _chunks.back()->add_segment(segment);
+    }
+  }
+
+  _chunks.back()->append(values);
 }
 
+bool Table::is_new_chunk_needed() { return _chunks.back()->size() >= _max_chunk_size; }
+
+uint16_t Table::column_count() const { return _column_names.size(); }
+
 uint64_t Table::row_count() const {
-  // Implementation goes here
-  return 0;
+  // Since we cannot delete data, all old chunks will have maximum chunk size
+  return ((_chunks.size() - 1) * _max_chunk_size) + _chunks.back()->size();
 }
 
-ChunkID Table::chunk_count() const {
-  // Implementation goes here
-  return ChunkID{0};
-}
+ChunkID Table::chunk_count() const { return (ChunkID)_chunks.size(); }
 
 ColumnID Table::column_id_by_name(const std::string& column_name) const {
-  // Implementation goes here
-  return ColumnID{0};
+  auto search = _column_name_mapping.find(column_name);
+  DebugAssert(search != _column_name_mapping.end(), "Could not find column_name");
+  return search->second;
 }
 
-uint32_t Table::max_chunk_size() const {
-  // Implementation goes here
-  return 0;
-}
+uint32_t Table::max_chunk_size() const { return _max_chunk_size; }
 
-const std::vector<std::string>& Table::column_names() const {
-  throw std::runtime_error("Implement Table::column_names()");
-}
+bool Table::is_empty() { return _chunks.size() == 1 && _chunks.front()->size() == 0; }
 
-const std::string& Table::column_name(ColumnID column_id) const {
-  throw std::runtime_error("Implement Table::column_name");
-}
+const std::vector<std::string>& Table::column_names() const { return _column_names; }
 
-const std::string& Table::column_type(ColumnID column_id) const {
-  throw std::runtime_error("Implement Table::column_type");
-}
+const std::string& Table::column_name(ColumnID column_id) const { return _column_names.at(column_id); }
+
+const std::string& Table::column_type(ColumnID column_id) const { return _column_types.at(column_id); }
+
+Chunk& Table::get_chunk(ChunkID chunk_id) { return *_chunks.at(chunk_id); }
+
+const Chunk& Table::get_chunk(ChunkID chunk_id) const { return *_chunks.at(chunk_id); }
+
+void Table::compress_chunk(ChunkID chunk_id) {
+  DebugAssert(chunk_id < _chunks.size(), "ChunkID out of range");
+  const auto chunk = _chunks[chunk_id];
+  auto compressed_chunk = std::make_shared<Chunk>();
+  // create new empty chunk
+  _chunks.push_back(compressed_chunk);
```

Comment:
why is the empty chunk already added to `_chunks` if it is later on moved to another position in `_chunks`?

Reply:
Yeah this did not make any sense. 

Fixed in next Commit